.travis.yml

```diff
@@ -15,16 +15,32 @@ before_install:
   - sudo apt-get update
   - sudo apt-get install -o Dpkg::Options::="--force-confold" -y docker-engine
 
+install:
+  - git clone https://github.com/docker-library/official-images.git ~/official-images
+
+before_script:
+  - image="$(awk '$1 == "FROM" { print $2; exit }' $VERSION/onbuild/Dockerfile)${VARIANT:+-$VARIANT}"
+  - version="$(awk '$2 == "PYTHON_VERSION" { print $3; exit }' $VERSION/${VARIANT:+$VARIANT/}Dockerfile)"
+  - echo "Building image $image with Python version $version"
+
 script:
-  - ./build.sh
+  - docker build -t "$image" "$VERSION${VARIANT:+/$VARIANT}"
+  # Run the "official-images" tests, but first tag our image like the official
+  # ones to get the same tests
+  - |
+    official_image="python:$VERSION${VARIANT:+-$VARIANT}"
+    docker tag "$image" "$official_image"
+    ~/official-images/test/run.sh "$official_image"
+  # the "onbuild" variant has to happen with the base variant because it's FROM it
+  - if [[ -z "$VARIANT" ]]; then docker build -t "${image}-onbuild" "$VERSION/onbuild"; fi
 
 after_script:
   - docker images
 
 before_deploy:
-  - pip install docker-ci-deploy
+  - docker login -u "$REGISTRY_USER" -p "$REGISTRY_PASS"
 deploy:
   provider: script
-  script: ./deploy.sh
+  script: ./deploy.py "$image" "$version" && if [[ -z "$VARIANT" ]]; then ./deploy.sh "${image}-onbuild" "$version"; fi
```

Comment:
I'm not hugely enthusiastic about this very long command line. Can we maybe have `deploy.py` handle that somehow?

Also, are we actually using `$VARIANT`? I don't see anything that sets it.